### ISSUE TITLE
Missing credentials for system proxy

### ISSUE THREAD

**Issue by Namue** [salvadordf/CEF4Delphi#265]
In my system there is a proxy configured which requires authentication.
When I visit a website in Chrome Browser (or other Browsers), I get an login dialog. After filled username / password, I get the website.
In the DCEF component I dont't get this login dialog and so I don't get the website.
I searched the web for a solution and found articles for CEF which said I should show my own dialog  for OnGetAuthCredentials. But when I tried this, the Event was not called. I found the cause in uCEFChromiumCore.TChromiumCore.doOnGetAuthCredentials: The Event is only called when isProxy is false. In my case isProxy is true and FProxyType is CEF_PROXYTYPE_SYSTEM, so nothing is called.
Should I make a merge request or should I use the component in another way?

**Comment by salvadordf** [comment]
Hi,

Unfortunately, I can't test that kind of proxy.

Try setting the proxy type as CEF_PROXYTYPE_FIXED_SERVERS and set all the TChromium properties related to the proxy settings manually.

Please, also try setting the proxy type as CEF_PROXYTYPE_SYSTEM and disabling the isProxy check in TChromiumCore.doOnGetAuthCredentials to see if you can use the callback with a username and password.

The MiniBrowser demo has all the code to set the proxy preferences in the top-right button.

**Comment by Namue** [comment]
When I set type to CEF_PROXYTYPE_FIXED_SERVERS and set the properties (ProxyScheme, ProxyServer, ProxyPort, ProxyByPassList, ProxyUsername, ProxyPassword) everything works like expected. But in my case I need to use the system settings.

When I set the type to CEF_PROXYTYPE_SYSTEM and disable the isProxy check in TChromiumCorte.doOnGetAuthCredentials my event procedure is called, so that everything works as expected.

I wouldn't only remove the isProxy check, because every request would return the proxylogin then. As I understand this method, it is also called when a website is requirering an authentication, where a popup would be needed even it the application uses an fixed proxy. I thing the isProxy check should be should be combined with the next check:

`
Result := False;
if isProxy and (FProxyType = CEF_PROXYTYPE_FIXED_SERVERS) and (callback <> nil) then
begin
  Result := True;
  callback.cont(FProxyUsername, FProxyPassword);
end else if Assigned(FOnGetAuthCredentials) then
  FOnGetAuthCredentials(Self, browser, originUrl, isProxy, host, port, realm, scheme, callback, Result);
`
